ocaml-lsp-server/src/ocamlformat_rpc.ml

```diff
@@ -0,0 +1,69 @@
+open Stdune
+open Import
+
+type process =
+  { pid : int
+  ; input : in_channel
+  ; output : out_channel
+  ; client : Ocamlformat_rpc_lib.client
+  }
+
+type state =
+  | Uninitialized
+  | Running of process
+  | Errored
+
+let running_process : state ref = ref Uninitialized
```

Comment:
This state should exist inside the LSP server instance.

Reply:
You mean that it should be a part of `State.t` ?

Reply:
Yes.